File: app/src/main/java/com/worldventures/dreamtrips/wallet/domain/entity/SmartCardDetails.java
package com.worldventures.dreamtrips.wallet.domain.entity;


import org.immutables.value.Value;

import java.util.Date;

@Value.Immutable
public interface SmartCardDetails {

   String serialNumber();

   long smartCardId();

   String bleAddress();

   String wvOrderId();

   String revVersion();

   String nxtOrderId();

   Date orderDate();

}

File: wallet/src/main/java/com/worldventures/wallet/domain/entity/SmartCardDetails.kt
package com.worldventures.wallet.domain.entity

import java.util.Date

data class SmartCardDetails(
      val deviceId: String?,
      // TODO Remove unused fields (they were used in SendWalletFeedbackCommand)
      val serialNumber: String = "",
      val bleAddress: String = "",
      val wvOrderId: String = "",
      val revVersion: String = "",
      val nxtOrderId: String = "",
      val orderDate: Date = Date())
